### PR TITLE
Curating Owners: cmd/kube-apiserver

### PR DESCRIPTION
**Pull request by apelisse**
cc @lavalamp @smarterclayton @krousey @nikhiljindal

In an effort to expand the existing pool of reviewers and establish a
two-tiered review process (first someone lgtms and then someone
experienced in the project approves), we are adding new reviewers to
existing owners files.

If You Care About the Process:
------------------------------

We did this by algorithmically figuring out who’s contributed code to
the project and in what directories.  Unfortunately, that doesn’t work
well: people that have made mechanical code changes (e.g change the
copyright header across all directories) end up as reviewers in lots of
places.

Instead of using pure commit data, we generated an excessively large
list of reviewers and pruned based on all time commit data, recent
commit data and review data (number of PRs commented on).

At this point we have a decent list of reviewers, but it needs one last
pass for fine tuning.

Also, see https://github.com/kubernetes/contrib/issues/1389.

TLDR:
-----

As an owner of a sig/directory and a leader of the project, here’s what
we need from you:

1. Use PR https://github.com/kubernetes/kubernetes/pull/35715 as an example.

2. The pull-request is made editable, please edit the `OWNERS` file to
remove the names of people that shouldn't be reviewing code in the future in
the **reviewers** section. You probably do NOT need to modify the **approvers**
section. Names asre sorted by relevance, using some secret statistics.

3. Notify me if you want some OWNERS file to be removed.  Being an
approver or reviewer of a parent directory makes you a reviewer/approver
of the subdirectories too, so not all OWNERS files may be necessary.

4. Please use ALIAS if you want to use the same list of people over and
over again (don't hesitate to ask me for help, or use the pull-request
above as an example)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36397)

### CODE REVIEW

**Review by krousey — APPROVED** [review]

**Comment by apelisse** [comment]
@krousey no names to remove from the list?

**Comment by krousey** [comment]
Huh. I was viewing an out of date version where the reviewers was empty. I will go back over this.

File: cmd/kube-apiserver/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/OWNERS]: Actually. Remove this.

File: cmd/kube-apiserver/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/OWNERS]: Remove.

File: cmd/kube-apiserver/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/OWNERS]: Remove.

File: cmd/kube-apiserver/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/OWNERS]: Remove.

File: cmd/kube-apiserver/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/OWNERS]: Remove this and everyone following except `ncdc`

File: cmd/kube-apiserver/app/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/app/OWNERS]: This should match the other file.

File: cmd/kube-apiserver/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/OWNERS]: Add @mml and @caesarxuchao

**Review by krousey — CHANGES_REQUESTED** [review]
I'm basing this mostly off the API machinery group.

**Comment by krousey** [comment]
And now reading the instructions.. I will go back and make the changes I requested.

File: cmd/kube-apiserver/app/OWNERS
Comment (apelisse) [inline cmd/kube-apiserver/app/OWNERS]: Or actually, we can just remove this file if we want to use the owners file from the parent directory

**Comment by apelisse** [comment]
Thanks @krousey!

File: cmd/kube-apiserver/app/OWNERS
Comment (krousey) [inline cmd/kube-apiserver/app/OWNERS]: api-machinery-{reviewers,approvers} ?

Comment (apelisse) [inline cmd/kube-apiserver/app/OWNERS]: Sounds good to me! I'll create the alias

**Comment by apelisse** [comment]
his name is missing from the list, but consider adding "AdoHe"

File: cmd/kube-apiserver/app/OWNERS
Comment (lavalamp) [inline cmd/kube-apiserver/app/OWNERS]: I vote remove this file and inherit from parent directory.

**Comment by apelisse** [comment]
@krousey @lavalamp I've made the changes and removed the sub-file. Please lgtm if you're ok with this.

**Comment by lavalamp** [comment]
/lgtm

@krousey you're still marked as wanting changes, not sure if the bot will care about that.

**Comment by apelisse** [comment]
@k8s-bot bazel test this

**Comment by apelisse** [comment]
@k8s-bot kops aws e2e test this

**Pull request merged**
